topics-and-advance-readings/psdad.md

```diff
@@ -0,0 +1,213 @@
+Alice intends each of the data fields in her message to be understood,
+in the context of the message.
+
+With **JSON-LD**, Alice might encode her message like this:
+```json
+{
+  "@context": {
+    "probe": "https://thermals.example/schema#probe",
+    "temp": "https://thermals.example/schema#temp"
+  },
+  "probe": 6, 
+  "temp": 34 
+}
+```
+
+Now, there is some in-band signaling of the semantics. The URIs act as
+global identifier (in a way "probe" and "temp" do not). And now Bob
+has the additional option of visiting https://thermals.example/schema
+in a browser and perhaps finding some useful information about the
+semantics of the data.  That sounds pretty good.
+
+These familiar techniques fall short, however, if there is confusion
+about which definitions to use. That confusion might come from an
+innocent mistake (such as poor coordination during evolution of the
+system), or it might come from a disinformation attack, with someone
+deliberately spreading misleading information about the semantics of
+Alice's data.
+
+As long as Alice and Bob know each other, are using the same software,
+or are using data formats defined by large standards organizations,
+the risk is probably small.  The risk becomes significant, however, in
+a dynamic market around an open data bus (such as the web), with many
+different components and systems passing data around, with
+increasingly complexity, and new formats or extensions appearing over
+time.
+
+Some people suggest the RDF (and XML) namespace architectures, as
+shown above in JSON-LD, solve this problem because one can use the
+namespace URL content as the master definition source. In theory this
+could work, but in practice it has three serious problems:
+
+1. The W3C RDF specifications and current practice do not actually
+give the namespace content any special privileged status. If the
+content disagrees with search engine results or word-of-mouth, it's
+unclear which should be taken as correct. It is also unclear whether
+there is any way to add such special status now, long after the specs
+have been published.
+
+2. This makes the namespace host a part of the critical security
+infrastructure, since it gets to say what the terms mean. Now, when
+Alice and Bob communicate, they must also trust the namespace host
+(thermal.example, in the above example). This could make it
+significantly harder to build secure data communication channels, and
+it increases the cost and liability for running a namespace
+host. Arguably, it increases centralization around major namespaces.
+
+3. Not everybody is willing to use RDF, even in the form of JSON-LD.
+
+These concerns are becoming more pressing as the [Credible Web
+Community Group](https://credweb.org) moves toward creating a data
+ecosystem to help combat online misinformation. There may be others in
+the larger community with similar concerns.
+
+## Alternatives
+
+I am not aware of anyone else having done work on this problem. I have
```

Comment:
@cwebber had mentioned this approach to us, quite possibly just relaying a conversation he had with you. I can't quite remember, but I think he's been playing around with this concept as well.

Reply:
Hmm, having now read the PSDAD spec, @cwebber's approach was different, in that he's cryptographically hashing the "vocab terms"... so, "controls and is the sole user of the website" transforms to SHA-256("controls and is the sole user of the website") and you use that value in the JSON-LD. For example (assuming multibase+multihash for the term, encoded in a JSON-LD document):

```
  // some magic context that denotes PSDAD encoding for terms
  "@context": " https://w3.org/2020/psdad/v1",
  "psdadSchema": "https://example.com/example-psdad-template",
  "z7vPPpikBuNHHbfpeyi3wN7f1L1nLC": "https://susan.example",
```